sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,41 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainModuleInitializer := {
+        if (scalaJSUseMainModuleInitializer.value) {
+          mainClass.value.fold {
+            throw new MessageOnlyException(
+                "No main class class detected, but " +
+                "scalaJSUseMainModuleInitializer was set to true. If you do " +
+                "not want to derive a main module initializer from " +
+                "discovered main classes, add " +
+                "`scalaJSMainModuleInitializer := None`")
+          } { mainClass =>
+            Some(ModuleInitializer.mainMethod(mainClass, "main"))
+          }
+        } else {
+          None
+        }
```

Comment:
Done

> Maybe to better align with sbt, we should map on mainClass instead and put the error message in the updated `scalaJSEntryPoints ++=`?

That's what I ended up doing, yes. In addition to being more aligned with sbt, it avoids the use of a dynTask, which is a nice win (dynTasks don't show up well in `inspect tree`).